### PR TITLE
Fix torch.jit.script breaking gradient tracking in normalizer

### PR DESCRIPTION
**Pull request by mcgibbon**
Fixes two GPU test failures.

`torch.jit.script` on `_normalize` and `_denormalize` caused autograd to silently drop `requires_grad` on `dict[str, Tensor]` outputs on CUDA after `autograd.grad` freed the computation graph. This broke the gradient chain in `SeparateRadiationStep` when `detach_radiation=False`.

`test_train_and_inference_with_derived_forcings` was flaky on GPU because `_setup` generated normalization stds from `np.random.randn()`, producing near-zero values that amplified inputs during normalization. Combined with SFNO + AMP bfloat16 + CUDA non-determinism, this occasionally caused NaN loss.

Changes:
- `fme.core.normalizer._normalize`, `fme.core.normalizer._denormalize`: remove `@torch.jit.script` decorators that caused autograd to lose gradient tracking on CUDA
- Remove unused `import torch.jit`
- `fme.ace.testing.fv3gfs_data.save_scalar_netcdf`: add `fill_value` parameter
- `fme.ace.test_train._setup`: replace manual seeding with `set_seed`; add `stats_std_fill_value` parameter to control normalization stds
- `fme.ace.test_train.test_train_and_inference_with_derived_forcings`: use `stats_std_fill_value=1.0` to avoid normalization amplification

- [x] Tests added
- [ ] If dependencies changed, "deps only" image rebuilt and "latest_deps_only_image.txt" file updated

### CODE REVIEW

**Review by mcgibbon — COMMENTED** [review]
Claude is convinced these shouldn't affect performance, and that there's no other way to fix the issue (including torch.compile, which also breaks these).

**Comment by mcgibbon** [comment]
The broken test was

```
        )[0]
        if detach_radiation:
            assert grad is None
        else:
>           assert grad is not None
E           assert None is not None

fme/core/step/test_radiation.py:159: AssertionError
========================================================== short test summary info ===========================================================
FAILED fme/core/step/test_radiation.py::test_detach_radiation[False] - assert None is not None
```

https://github.com/ai2cm/ace/actions/runs/25696313514/job/75445328882?pr=1095

**Comment by elynnwu** [comment]
> The broken test was
> 
> ```
>         )[0]
>         if detach_radiation:
>             assert grad is None
>         else:
> >           assert grad is not None
> E           assert None is not None
> 
> fme/core/step/test_radiation.py:159: AssertionError
> ========================================================== short test summary info ===========================================================
> FAILED fme/core/step/test_radiation.py::test_detach_radiation[False] - assert None is not None
> ```
> 
> https://github.com/ai2cm/ace/actions/runs/25696313514/job/75445328882?pr=1095

Can confirm that this fixes this test. The failing CI test is a different test (`fme/ace/test_train.py::test_train_and_inference_with_derived_forcings`) which I'm not able to reproduce.

**Comment by mcgibbon** [comment]
Probably a flake, I'll re-run the tests.

**Comment by elynnwu** [comment]
Is it strange that tests aren't failing in main?

**Comment by mcgibbon** [comment]
> Is it strange that tests aren't failing in main?

Yes, but it's been a long time since I've stopped being surprised by flaky GPU failure behavior.

**Review by elynnwu — APPROVED** [review]
